front/src/app/todo-list/todo-list.component.html

```diff
@@ -1,32 +1,78 @@
 <div [formGroup]="radioBtnGroup" class="task-list">
   <mat-radio-group formControlName="activeTaskList" class="task-list__group">
     <a
-      *ngFor="let todoList of allTaskLists?.lists" 
+      *ngFor="let todoList of allTaskLists?.lists; let taskListIndex = index" 
       [routerLink]="['/todo']"
       [queryParams]="{list: todoList.id}"
       class="task-list__link"
     >
-      <mat-radio-button [value]="todoList.id" class="task-list__list">{{todoList.title}}</mat-radio-button>
+      <mat-radio-button [value]="todoList.id" class="task-list__list">
+        <div class="task-list__task-group">
+          <span *ngIf="selectedTaskGroupIndex !== taskListIndex">
+            {{todoList.title}}
+          </span>
+          <input
+            [ngClass]="{'hide-element': selectedTaskGroupIndex !== taskListIndex}"
+            #renameTaskGroupFormField
+            matInput
+            name="input{{taskListIndex}}"  
+            type="text"
+            (keydown.enter)="renameTaskGroup(todoList.id, renameTaskGroupFormField.value)"
+            (blur)="onRenameTaskGroupInputBlur()"
+            (click)="stopPropagation($event); false"

```

Comment:
it's quite ugly to write `false` right in the html click handler (and i never seen such approach)

Do we really need that? please check